src/Conways.GameOfLife.API/Program.cs

```diff
@@ -40,20 +40,24 @@
     })
     .EnableApiVersionBinding();
 
-builder.Services.AddEndpointsApiExplorer()
-    .AddSwaggerGen(options =>
-    {
-        options.SwaggerDoc("v1",
-            new OpenApiInfo { Title = "Conway's Game Of Life API", Version = "v1" });
-    })
-    .AddOpenTelemetryObservability(builder.Configuration);
+builder.Services.AddEndpointsApiExplorer();
+
+builder.Services.AddOpenApi();
+
+builder.Services.AddOpenTelemetryObservability(builder.Configuration);
 
 var app = builder.Build();
 
-if (app.Environment.IsDevelopment())
+
+var documentationEnabled = app.Configuration.GetValue<bool>("ApiDocumentation:Enabled");
+
+if (documentationEnabled)
 {
-    app.UseSwagger();
-    app.UseSwaggerUI();
+    
+    app.MapOpenApi();
+    
+    
+    app.MapScalarApiReference();
```

Comment:
Pode ajustar a formatação, tem algumas linhas em branco à mais aqui
```suggestion    
    app.MapOpenApi();
    app.MapScalarApiReference();
```

Reply:
Ajustei a formatação e removi as linhas em branco